lib/cocoapods/generator/xcconfig/xcconfig_helper.rb

```diff
@@ -103,15 +103,23 @@ def self.add_settings_for_file_accessors_of_target(aggregate_target, pod_target,
         #
         def self.add_static_dependency_build_settings(aggregate_target, pod_target, xcconfig, file_accessor)
           file_accessor.vendored_static_frameworks.each do |vendored_static_framework|
-            if aggregate_target.nil? || aggregate_target.target_definition.inheritance != 'search_paths'
-              XCConfigHelper.add_framework_build_settings(vendored_static_framework, xcconfig, pod_target.sandbox.root)
-            end
+            should_add_other_ldflags = XCConfigHelper.should_link_static_framework_or_library(aggregate_target, pod_target)
+            XCConfigHelper.add_framework_build_settings(vendored_static_framework, xcconfig, pod_target.sandbox.root, should_add_other_ldflags)
           end
           file_accessor.vendored_static_libraries.each do |vendored_static_library|
-            XCConfigHelper.add_library_build_settings(vendored_static_library, xcconfig, pod_target.sandbox.root)
+            should_add_other_ldflags = XCConfigHelper.should_link_static_framework_or_library(aggregate_target, pod_target)
+            XCConfigHelper.add_library_build_settings(vendored_static_library, xcconfig, pod_target.sandbox.root, should_add_other_ldflags)
           end
         end
 
+        # TODO
+        #
+        def self.should_link_static_framework_or_library(aggregate_target, pod_target)
+          return true if aggregate_target.nil? || aggregate_target.target_definition.inheritance != 'search_paths'
+          return aggregate_target.pod_targets.include?(pod_target)
```

Comment:
@benasher44 one of the flaws this logic has is that transitive dependencies with pre-built frameworks/libraries are still included in the `other_ldflags`. I might need a bit of help to tackle this.